File: app/src/main/java/com/fishweeb/practical/BackEntity.java
package com.fishweeb.practical;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.util.DisplayMetrics;
import android.view.SurfaceView;

public class BackEntity extends ButtonEntity
{
    private Bitmap bmpPress = null;
    private Bitmap bmpOriginal = null;
    private String nextState;
    @Override
    public void Init(SurfaceView _view) {
        // Define anything you need to use here
        bmpOriginal = Bitmap.createScaledBitmap(ImageManager.Instance.GetImage(IMAGE.I_BACKBUTTON),(int)width,(int)height,true);
        bmpPress = Bitmap.createScaledBitmap(ImageManager.Instance.GetImage(IMAGE.I_BACKBUTTON_PRESSED),(int)width,(int)height,true);
        bmp = bmpOriginal;

        velocity.x = 0;
        velocity.y = 0;

        if (width > height)
            radius = width*0.5f;
        else
            radius = height*0.5f;
    }

    @Override
    public void OnClickFunction()
    {
        bmp = bmpPress;
    }

    @Override
    public void OffClickFunction()
    {
        bmp = bmpOriginal;
        StateManager.Instance.ChangeState(nextState);
    }

    public static BackEntity Create(float xPos,float yPos,float _width,float _height,String _nextState)
    {
        BackEntity result = new BackEntity();
        result.Pos.x = xPos;
        result.Pos.y = yPos;
        result.width = _width;
        result.height = _height;
        result.nextState = _nextState;
        EntityManager.Instance.AddEntity(result);
        return result;
    }
}

File: app/src/main/java/edu/androidgroupproject/BackEntity.kt
package edu.androidgroupproject

import android.graphics.Bitmap
import android.view.SurfaceView

class BackEntity : ButtonEntity() {
    private var bmpPress: Bitmap? = null
    private var bmpOriginal: Bitmap? = null
    private var nextState: String? = null
    override fun Init(_view: SurfaceView) {
        // Define anything you need to use here
        bmpOriginal = Bitmap.createScaledBitmap(
            ImageManager.Instance.GetImage(IMAGE.I_BACKBUTTON)!!,
            width.toInt(),
            height.toInt(),
            true
        )
        bmpPress = Bitmap.createScaledBitmap(
            ImageManager.Instance.GetImage(IMAGE.I_BACKBUTTON_PRESSED)!!,
            width.toInt(),
            height.toInt(),
            true
        )
        bmp = bmpOriginal

        velocity.x = 0f
        velocity.y = 0f

        radius = if (width > height) width * 0.5f
        else height * 0.5f
    }

    override fun OnClickFunction() {
        bmp = bmpPress
    }

    override fun OffClickFunction() {
        bmp = bmpOriginal
        StateManager.Instance.ChangeState(nextState)
    }

    companion object {
        fun Create(
            xPos: Float,
            yPos: Float,
            _width: Float,
            _height: Float,
            _nextState: String?
        ): BackEntity {
            val result = BackEntity()
            result.Pos.x = xPos
            result.Pos.y = yPos
            result.width = _width
            result.height = _height
            result.nextState = _nextState
            EntityManager.Instance.AddEntity(result)
            return result
        }
    }
}
